packages/core/content-manager/server/services/entity-manager.js

```diff
@@ -72,40 +73,77 @@ module.exports = ({ strapi }) => ({
     return assoc(`${CREATED_BY_ATTRIBUTE}.roles`, roles, entity);
   },
 
-  find(opts, uid) {
+  /**
+   * Extend this function from other plugins to add custom mapping of entity
+   * responses
+   * @param {Object} entity
+   * @returns
+   */
+  mapEntity(entity) {
+    return entity;
+  },
+
+  /**
+   * Some entity manager functions may return multiple entities or one entity.
+   * This function maps the response in both cases
+   * @param {Array|Object|null} entities
+   * @param {string} uid
+   */
+  async mapEntitiesResponse(entities, uid) {
+    if (entities?.results) {
+      const mappedResults = await mapAsync(entities.results, (entity) =>
+        this.mapEntity(entity, uid)
+      );
+      return { ...entities, results: mappedResults };
+    }
+    // if entity is single type
```

Comment:
Could you check that the type of entities is single type?

Reply:
That make sense,
Just a point, that would require to load the whole uid model in this function. 
We thought it was a bit of an overhead , but it makes sense to be more explicit here. wdyt @jhoward1994 ?

Reply:
I think safety for single types is already covered in the entity service. see:

https://github.com/strapi/strapi/blob/19e9f952efb08c9dca8f261132479a239d7659d8/packages/core/strapi/lib/services/entity-service/index.js#L85

However we could change this code block to 
```
    if (entities?.results || Array.isArray(entities)) {
      const mappedResults = await mapAsync(entities?.results || entities, (entity) =>
        this.mapEntity(entity, uid)
      );
      return { ...entities, results: mappedResults };
    }
    // if entity is single type
    return this.mapEntity(entities, uid);
```

wdyt?